### ISSUE TITLE
Service exited with abnormal code: 1

### ISSUE THREAD

**Issue by digitalmoksha** [zorgiepoo/sparkle-ui-xpc-issues#43]
I'm breaking my head.  I keep getting this error in the system.log:

```
(com.digitalmoksha.versatil-sparkle-progress[675]): Service exited with abnormal code: 1
(com.digitalmoksha.versatil-sparkle-updater[674]): Service exited with abnormal code: 1
```

It get's all the way through extracting the update, then throws up the "Update Error!" dialog, and the above entries appear in the log.  I thought it might be delta processing, but I've removed the deltas.

Any ideas what might be causing this?

**Comment by zorgiepoo** [comment]
Well, I would like to believe there would be more logs or information. Have Console.app running before running into the error and perhaps filter for `Autoupdate` or `Updater.app` or your own application or for `sparkle` (which would include the InstallerLauncher xpc service). I guess the InstallerLauncher XPC service needs to be signed properly too. It's one of the services that isn't supposed to be sandboxed (i.e, no entitlements; along with InstallerConnection and InstallerStatus), even if your application is sandboxed.

**Comment by digitalmoksha** [comment]
Unfortunately, that's all the log information there was - I already tried the things you mentioned.

But I figured it out.  The culprit is the `generate_appcast` tool, and relates to [#959](https://github.com/sparkle-project/Sparkle/issues/959).  An incorrect appcast was generated because a replaced archive doesn't get re-expanded.

It would have been nice to have a little better error information in the log file (archive size mismatch or corruption, or whatever) but maybe it's not possible at the level it was happening.
